biz.aQute.bndlib/src/aQute/bnd/osgi/resource/CapReq.java

```diff
@@ -20,26 +22,27 @@
 	private final Resource				resource;
 	private final Map<String,String>	directives;
 	private final Map<String,Object>	attributes;
+	private transient int				hashCode	= 0;
```

Comment:
The class can be marked immutable since the hashCode field is transient and always recomputes the same value. There is no need for this feild to be final or even visible across threads :-)